Assets/Scripts/Release/MassReleaseStrategyBase.cs

```diff
@@ -16,62 +16,67 @@ public MassReleaseStrategyBase(IAgent agent, IAssignment assignment) : base(agen
-      Vector3 position = Agent.Position;
-      Simulation.CartesianCoordinates positionCoordinates = Coordinates3.ToProto(position);
+    // Release all sub-interceptors.
+    Configs.SubAgentConfig subAgentConfig = carrier.AgentConfig.SubAgentConfig;
+    Vector3 position = carrier.Position;
+    Simulation.CartesianCoordinates positionCoordinates = Coordinates3.ToProto(position);
 
-      Vector3 velocity = Agent.Velocity;
-      Vector3 perpendicularDirection = Vector3.Cross(velocity, Vector3.up);
+    Vector3 velocity = carrier.Velocity;
+    Vector3 perpendicularDirection = Vector3.Cross(velocity, Vector3.up);
 
-      for (int i = 0; i < subAgentConfig.NumSubAgents; ++i) {
-        // Fan the submunitions radially outwards from the carrier's velocity vector.
-        Vector3 lateralDirection =
-            Quaternion.AngleAxis(i * 360 / subAgentConfig.NumSubAgents, velocity) *
-            perpendicularDirection;
-        Simulation.CartesianCoordinates velocityCoordinates =
-            Coordinates3.ToProto(Vector3.RotateTowards(
-                velocity, lateralDirection, maxRadiansDelta: _fanOutAngleDegrees * Mathf.Deg2Rad,
-                maxMagnitudeDelta: Mathf.Cos(_fanOutAngleDegrees * Mathf.Deg2Rad)));
-        Simulation.State initialState = new Simulation.State() {
-          Position = positionCoordinates,
-          Velocity = velocityCoordinates,
-        };
-        IAgent subInterceptor =
-            SimManager.Instance.CreateInterceptor(subAgentConfig.AgentConfig, initialState);
-        if (subInterceptor != null) {
-          if (subInterceptor.Movement is MissileMovement movement) {
-            movement.FlightPhase = Simulation.FlightPhase.Boost;
-          }
-          releasedAgents.Add(subInterceptor);
+    var releasedAgents = new List<IAgent>();
+    for (int i = 0; i < subAgentConfig.NumSubAgents; ++i) {
+      // Fan the submunitions radially outwards from the carrier's velocity vector.
+      Vector3 lateralDirection =
+          Quaternion.AngleAxis(i * 360 / subAgentConfig.NumSubAgents, velocity) *
+          perpendicularDirection;
+      Simulation.CartesianCoordinates velocityCoordinates =
+          Coordinates3.ToProto(Vector3.RotateTowards(
+              velocity, lateralDirection, maxRadiansDelta: _fanOutAngleDegrees * Mathf.Deg2Rad,
+              maxMagnitudeDelta: Mathf.Cos(_fanOutAngleDegrees * Mathf.Deg2Rad)));
+      Simulation.State initialState = new Simulation.State() {
+        Position = positionCoordinates,
+        Velocity = velocityCoordinates,
+      };
+      IAgent subInterceptor =
+          SimManager.Instance.CreateInterceptor(subAgentConfig.AgentConfig, initialState);
+      if (subInterceptor != null && subInterceptor is IInterceptor subInterceptorInterceptor) {
+        if (subInterceptor.Movement is MissileMovement movement) {
+          movement.FlightPhase = Simulation.FlightPhase.Boost;
         }
+        releasedAgents.Add(subInterceptor);
       }
```

Comment:
Why does `subInterceptorInterceptor` declared and type checked here? Was it meant to be used? Right now its declared but ignored. Confusing, maybe some incomplete refactoring

Reply:
Good catch.